### PR TITLE
feat(HACBS-402): add conventional-commit checks

### PR DESCRIPTION
**Pull request by johnbieren**
Add conventional-commit checking that will be run on commit messages
that are added in new pull requests. Run the check as part of pr github
action.

Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

**Comment by johnbieren** [comment]
I took all the rules product CI is using

File: .gitlint [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[general]
+ignore=T5
```
Comment (davidmogar) [inline .gitlint:2]: Could you add `contrib=contrib-title-conventional-commits,CC1` here and remove it from the action?
```suggestion
contrib=contrib-title-conventional-commits,CC1
ignore=T5
```

File: .gitlint [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+[title-max-length]
+line-length=72
```
Comment (davidmogar) [inline .gitlint:5]: Both this and the next one are the default values, but yeah, probably better to make it clear.

File: .gitlint [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+[body-max-line-length]
+line-length=72
+
+[body-first-line-empty]
+
+[contrib-title-conventional-commits]
```
Comment (davidmogar) [inline .gitlint:12]: Missing new line at the end.

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -100,3 +100,16 @@ jobs:
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install gitlint into container
+        run: python -m pip install gitlint
+      - name: Run gitlint check
+        run: gitlint --config .gitlint --contrib contrib-title-conventional-commits,CC1 --commits origin/${{ github.event.pull_request.base.ref }}..HEAD
```
Comment (davidmogar) [inline .github/workflows/pr.yml:115]: Missing new line at the end.

Also, as per the comment below:
```suggestion
        run: gitlint --commits origin/${{ github.event.pull_request.base.ref }}..HEAD
```
I also removed the `--config` option as we are using the default.

**Review by davidmogar — COMMENTED** [review]
Neat change! Just some minor comments.

**Review by davidmogar — APPROVED** [review]

**Pull request merged**